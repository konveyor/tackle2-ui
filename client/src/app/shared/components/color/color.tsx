import React from "react";
import { useTranslation } from "react-i18next";

import { Split, SplitItem } from "@patternfly/react-core";

import { DEFAULT_COLOR_LABELS } from "@app/Constants";
import "./color.css";

export interface ColorProps {
  hex: string;
}

export const Color: React.FC<ColorProps> = ({ hex }) => {
  const { t } = useTranslation();

  const colorName = DEFAULT_COLOR_LABELS.get(hex.toLowerCase());

  return (
    <Split hasGutter>
      <SplitItem>
        <div
          className="color"
          style={{ backgroundColor: hex }}
          cy-data="color-box"
        ></div>
      </SplitItem>
      <SplitItem isFilled>
        <span cy-data="color-label">
          {colorName ? t(`colors.${colorName}`) : hex}
        </span>
      </SplitItem>
    </Split>
  );
};
