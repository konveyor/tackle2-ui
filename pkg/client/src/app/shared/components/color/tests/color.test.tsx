import React from "react";
import { shallow, mount } from "enzyme";
import { global_palette_blue_300 as blue } from "@patternfly/react-tokens";
import { Color } from "../color";

describe("Color", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(<Color hex="#FFF" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Applies hex color style", () => {
    const wrapper = mount(<Color hex="#FFF" />);

    expect(
      wrapper.find('div[cy-data="color-box"]').prop("style")
    ).toHaveProperty("backgroundColor", "#FFF");
  });

  it("Applies color label with lowercase hex", () => {
    const wrapper = mount(<Color hex={blue.value.toLowerCase()} />);

    expect(wrapper.find('span[cy-data="color-label"]').text()).toEqual(
      "colors.blue"
    );
  });

  it("Applies color label with upercase hex", () => {
    const wrapper = mount(<Color hex={blue.value.toUpperCase()} />);

    expect(wrapper.find('span[cy-data="color-label"]').text()).toEqual(
      "colors.blue"
    );
  });

  it("Applies color hex code", () => {
    const wrapper = mount(<Color hex={"#fff"} />);

    expect(wrapper.find('span[cy-data="color-label"]').text()).toEqual("#fff");
  });
});
