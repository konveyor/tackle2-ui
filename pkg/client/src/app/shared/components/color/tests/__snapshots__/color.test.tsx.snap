// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Renders without crashing 1`] = `
<Split
  hasGutter={true}
>
  <SplitItem>
    <div
      className="color"
      cy-data="color-box"
      style={
        Object {
          "backgroundColor": "#FFF",
        }
      }
    />
  </SplitItem>
  <SplitItem
    isFilled={true}
  >
    <span
      cy-data="color-label"
    >
      #FFF
    </span>
  </SplitItem>
</Split>
`;
